src/engine/strat_engine/crypt/shared.rs

```diff
@@ -838,20 +484,44 @@ pub fn activate(
             )));
         }
     }
-    log_on_failure!(
+    if let Some(p) = passphrase {
```

Comment:
The particular logical arrangement right here leads to an undesirable consequence:

If a passphrase is given _and_ it is incorrect the pool or device as the case may be will not be opened, _even_ if the correct key is in the keyring (in the case where the pool is encrypted with a key), or the equivalent notion if the pool was encrypted using Clevis.

The problem looks like this. In each case, the pool is encrypted only with the method it is expected to use when being started and in each case, a passphrase is specified but it is incorrect. I think we agreed that the passphrase would act as a fallback, in case the method specified failed without it.

```
$ sudo stratis pool start
--name spool1 --unlock-method keyring --capture-key
Enter passphrase followed by the return key:
Verify passphrase entered:
Execution failed:
stratisd failed to perform the operation that you requested. It
returned the following information via the D-Bus: ERROR: Cryptsetup
error: IO error occurred: Operation not permitted (os error 1).

$ sudo stratis pool start
--name spool2 --unlock-method keyring --capture-key
Enter passphrase followed by the return key:
Verify passphrase entered:
Execution failed:
stratisd failed to perform the operation that you requested. It
returned the following information via the D-Bus: ERROR: An attempt to
set up pool with UUID 6468c4db-fea9-4b75-8450-80dd1f051033 from the
assembled devices failed; Cryptsetup error: IO error occurred:
Operation not permitted (os error 1).

$ sudo stratis pool start
--name spool3 --unlock-method clevis --capture-key
Enter passphrase followed by the return key:
Verify passphrase entered:
Execution failed:
stratisd failed to perform the operation that you requested. It
returned the following information via the D-Bus: ERROR: Cryptsetup
error: IO error occurred: Operation not permitted (os error 1).

$ sudo stratis pool start
--name spool4 --unlock-method clevis --capture-key
Enter passphrase followed by the return key:
Verify passphrase entered:
Execution failed:
stratisd failed to perform the operation that you requested. It
returned the following information via the D-Bus: ERROR: An attempt to
set up pool with UUID 78389678-0e95-411a-b71a-81d2e7c64649 from the
assembled devices failed; Cryptsetup error: IO error occurred:
Operation not permitted (os error 1).
```

Reply:
I'm actually interested in discussing this in stand up. I did that somewhat intentionally only because let's say we rework boot from root work where it always prompts for a passphrase, for example. Is it more or less desirable to enter the wrong passphrase and have it still unlock because of Clevis or to fail because you entered the wrong passphrase?

Reply:
Resolved in stand up.

Reply:
I think we decided that this is actually the behavior we want.

Reply:
This is the desired behavior, otherwise the user might be mislead into believing that an incorrect passphrase is actually correct.